src/c++/perf_analyzer/client_backend/client_backend.h

```diff
@@ -558,6 +558,15 @@ class InferInput {
   virtual Error SetSharedMemory(
       const std::string& name, size_t byte_size, size_t offset = 0);
 
+  /// Get access to the buffer holding raw input. Note the buffer is owned by
+  /// InferInput instance. Users can copy out the data if required to extend
+  /// the lifetime.
+  /// \param buf Returns the pointer to the start of the buffer.
+  /// \param byte_size Returns the size of buffer in bytes.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  virtual Error RawData(const uint8_t** buf, size_t* byte_size);
```

Comment:
I assume you discussed this with Iman?

Reply:
Yeah, we discussed two main approaches for this: either get the input data at the time when PA is passing the data to `InferInput` object or fetch it later from `InferInput` object. The former is happening inside the `InferDataManager` which is pretty far from where it's needed so I decided to go with the latter approach by adding this interface to the class.

viz @Tabrizian